src/Sructs.c

```diff
@@ -0,0 +1,40 @@
+/*
+ Created by Elchairoy Meir on 17/11/2021.
+ Definition of table, column a cell structs.
+*/
+
+#define MAX_COLUMN_NAME_LEN 30
+#define MAX_TABLE_NAME_LEN 30
+
+enum types {Character = 0,Integer,Double,String}; /* All the possible types of data in each column/cell. */
+
+/* The struct of cells: */
+typedef struct
+{
+    enum types type;
+    unsigned index; /* The index of the cell in the initial column (after the column goes through changes that
```

Comment:
I don't think that it's needed. We should discuss it.
And besides, if it is an index - it should be much bigger than unsigned!

Reply:
That is definitely required, if for example the user wants to sort the table by one column, we can sort only the spesific column and yet the initial rows will remain the same.